SLHCUpgradeSimulations/Configuration/python/customise_TkRecoMaterial.py

```diff
@@ -0,0 +1,20 @@
+'''Customization functions for cmsDriver to change the phase-2 tracker reco material for geometry D49'''
+import FWCore.ParameterSet.Config as cms
+def customizeRecoMaterial(process):
+    
+    ''' will replace one tracker reco material file with another one for geometry D49
+    syntax: --customise SLHCUpgradeSimulations/Configuration/customise_TkRecoMaterial.customizeRecoMaterial
+    '''
+    
+    if hasattr(process,'XMLIdealGeometryESSource') and hasattr(process.XMLIdealGeometryESSource,'geomXMLFiles'):
+        
+        print("customizing Tracker Reco Material")
+        process.XMLIdealGeometryESSource.geomXMLFiles.remove(
```

Comment:
you should probably check if this file is present in the list before removing it, otherwise python will raise a ValueError

Reply:
done, thanks for the comment

Reply:
it's a rather non-pythonic fix though and also the new code allows for silent failing of the application of the customization function if the file is not present.

I had in mind something like:
```diff
diff --git a/SLHCUpgradeSimulations/Configuration/python/customise_TkRecoMaterial.py b/SLHCUpgradeSimulations/Configuration/python/customise_TkRecoMaterial.py
index 8d99be95dad..75e64ce891f 100644
--- a/SLHCUpgradeSimulations/Configuration/python/customise_TkRecoMaterial.py
+++ b/SLHCUpgradeSimulations/Configuration/python/customise_TkRecoMaterial.py
@@ -1,20 +1,21 @@
 '''Customization functions for cmsDriver to change the phase-2 tracker reco material for geometry D49'''
 import FWCore.ParameterSet.Config as cms
-def customizeRecoMaterial(process):
-    
+def customizeTkRecoMaterialD49(process):
+
     ''' will replace one tracker reco material file with another one for geometry D49
-    syntax: --customise SLHCUpgradeSimulations/Configuration/customise_TkRecoMaterial.customizeRecoMaterial
+    syntax: --customise SLHCUpgradeSimulations/Configuration/customise_TkRecoMaterial.customizeTkRecoMaterialD49
     '''
-    
+
     if hasattr(process,'XMLIdealGeometryESSource') and hasattr(process.XMLIdealGeometryESSource,'geomXMLFiles'):
-        
-        print("customizing Tracker Reco Material")
-        process.XMLIdealGeometryESSource.geomXMLFiles.remove(
-            'Geometry/TrackerRecoData/data/PhaseII/TiltedTracker613_MB_2019_04/trackerRecoMaterial.xml'
-        )
-        
-        process.XMLIdealGeometryESSource.geomXMLFiles.append(
-            'Geometry/TrackerRecoData/data/PhaseII/TiltedTracker613_MB_2019_04/v2_ITonly/trackerRecoMaterial.xml'
-        )
-        
+
+        try:
+            process.XMLIdealGeometryESSource.geomXMLFiles.remove(
+                'Geometry/TrackerRecoData/data/PhaseII/TiltedTracker613_MB_2019_04/trackerRecoMaterial.xml'
+            )
+            process.XMLIdealGeometryESSource.geomXMLFiles.append(
+                'Geometry/TrackerRecoData/data/PhaseII/TiltedTracker613_MB_2019_04/v2_ITonly/trackerRecoMaterial.xml'
+            )
+        except ValueError:
+            raise SystemExit("\n\n ERROR! Could not replace trackerRecoMaterial.xml file, please check check if D49 geometry is being used \n\n")
+
     return process

```

Reply:
ciao
thanks for the comment, but the `raise SystemExit` will make the code exiting, is it what we want ?

Reply:
yes, as the customization should simply NOT be run if the geometry is not the correct one.